OpenSim/Common/DelimFileAdapter.h

```diff
@@ -400,39 +399,62 @@ DelimFileAdapter<T>::extendRead(const std::string& fileName) const {
                      _timeColumnLabel,
                      column_labels[0]);
     column_labels.erase(column_labels.begin());
-    // Set the column labels as metadata.
-    ValueArray<std::string> value_array{};
-    for(const auto& cl : column_labels)
-        value_array.upd().push_back(SimTK::Value<std::string>{cl});
-    typename TimeSeriesTable_<T>::DependentsMetaData dep_metadata{};
-    dep_metadata.setValueArrayForKey("labels", value_array);
-    table->setDependentsMetaData(dep_metadata);
 
     // Read the rows one at a time and fill up the time column container and
-    // the data container.
+    // the data container. Start with a reasonable initial capacity for
+    // tradeoff between a small file and larger files. 100 worked well for
+    // a 50 MB file with ~80000 lines.
+    std::vector<double> timeVec;
+    int initCapacity = 100;
+    int ncol = static_cast<int>(column_labels.size());
+    timeVec.reserve(initCapacity);
+    SimTK::Matrix_<T> matrix(initCapacity, ncol);
+    
+    // Initialize current row and capacity
+    int curCapacity = initCapacity;
+    int curRow = 0;
```

Comment:
Minor, but *curRow* is not the current row but the  size, that is why  you have to set the `curRow-1`th row in the Matrix.  I found that a bit confusing.